src/Common/FileCache.cpp

```diff
@@ -379,30 +553,27 @@ LRUFileCache::FileSegmentCell * LRUFileCache::addCell(
         FileSegment::State result_state = state;
         if (state == FileSegment::State::EMPTY && enable_cache_hits_threshold)
         {
-            auto record = records.find({key, offset});
+            auto record = stash_records.find({key, offset});
 
-            if (record == records.end())
+            if (record == stash_records.end())
             {
-                auto queue_iter = stash_queue.add(key, offset, 0, cache_lock);
-                records.insert({{key, offset}, queue_iter});
+                auto priority_iter = stash_priority->add(key, offset, 0, cache_lock);
+                stash_records.insert({{key, offset}, priority_iter});
 
-                if (stash_queue.getElementsNum(cache_lock) > max_stash_element_size)
+                if (stash_priority->getElementsNum(cache_lock) > max_stash_element_size)
                 {
-                    auto remove_queue_iter = stash_queue.begin();
-                    records.erase({remove_queue_iter->key, remove_queue_iter->offset});
-                    stash_queue.remove(remove_queue_iter, cache_lock);
+                    auto remove_priority_iter = stash_priority->getNewIterator(cache_lock)->getWriteIterator();
```

Comment:
May be just make it possible to do `stash_priority->getWriteIterator`? This  `getNewIterator()->getWriteIterator` does not look nice. I am in doubt about the idea that we can get WriteIterator from ReadIterator.